src/recordmerge.jl

```diff
@@ -0,0 +1,13 @@
+"""
+    recordmerge(record; fields_to_merge...)
```

Comment:
The symmetry with `rowmerge` is appealing but this is technically not a style guide compliant name. `record_merge` would work. Though would it make sense for this to instead be a method of `rowmerge` or even `merge`?

Reply:
> Though would it make sense for this to instead be a method of rowmerge or even merge?

i think this would make sense as a `rowmerge` method actually (should be docstring'd especially for the `AbstractRecord` method, of course)